core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -62,63 +61,18 @@
   /**
    * Returns a new {@link ManifestReader} for an {@link InputFile}.
    * <p>
-   * <em>Note:</em> Most callers should use {@link #read(ManifestFile, FileIO, Map)} to ensure that
+   * <em>Note:</em> Callers should use {@link ManifestFiles#read(ManifestFile, FileIO, Map)} to ensure that
    * manifest entries with partial metadata can inherit missing properties from the manifest metadata.
-   * <p>
-   * <em>Note:</em> Most callers should use {@link #read(InputFile, Map)} if all manifest entries
-   * contain full metadata and they want to ensure that the schema used by filters is the latest
-   * table schema. This should be used only when reading a manifest without filters.
    *
    * @param file an InputFile
    * @return a manifest reader
+   * @deprecated use {@link ManifestFiles#read(ManifestFile, FileIO, Map)}.
    */
+  @Deprecated
```

Comment:
Why do we keep only this version for compatibility? I thought the one below was the preferred way. Would it make sense to keep both and mark them deprecated? Versions that accept `ManifestFile` were not released yet, so we should definitely remove those.

Reply:
Looks like `read(InputFile, Map)` was also released in 0.7.0. I kept this one because I know we have people using it, but we can keep the other one as well. Do you think we should?

Reply:
The version released in 0.7.0 was `read(InputFile, Function)` because we used to pass a function to look up a spec instead of a map. We could bring that back and deprecate it, but I think I'm fine with just deprecating this one. This is the one that I know people use.

Reply:
I think it makes sense for consistency. We were using it but we will have to adapt it anyway.

Reply:
For us, it does not matter.

Reply:
Nevermind, it was much easier than I thought to bring this back because we don't pass the map to any other class.